src/aiovodafone/api.py

```diff
@@ -526,19 +542,25 @@ async def login(self) -> bool:
         await self._set_cookie()
         await self._reset()
 
-        # First  try with both  username and password encrypted
-        # Second try with plain username and password encrypted
-        try:
-            _LOGGER.debug("Login first try: username[encrypted], password[encrypted]")
-            logged = await self._login_json(
-                await self._encrypt_string(self.username),
-                await self._encrypt_string(self.password),
-            )
-        except CannotAuthenticate:
-            _LOGGER.debug("Login second try: username[plain], password[encrypted]")
-            logged = await self._login_json(
-                self.username, await self._encrypt_string(self.password)
-            )
+        if not self.encryption_key:
+            _LOGGER.debug("Login: username[plain], password[challenge encrypted]")
+            logged = await self._login_json(self.username, self.password, True)
```

Comment:
you can leave the code as it is and use "self.encryption_key" as third parameter of the _login_json instead